### PR TITLE
feat(react-query): add cli status result table

### PR DESCRIPTION
**Pull request by gwansikk**
# Overview

close: #1029 

I used the [cli-table3](https://github.com/cli-table/cli-table3) package to implement this.

| When the version is correct |  When the version is incorrect |
| --- | --- |
|image|image|

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: packages/react-query/src/scripts/utils/table.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  const table = new Table({
+    head: ['', 'result', 'status', 'advice'],
+    style: { head: [] },
+  })
+
+  console.log(tanStackReactQueryMajorVersion)
+
+  table.push(['version', `@suspensive/react-query@${packageJson.version}`, '🟢', ''])
+  table.push(['export', `@suspensive/react-query-${currentVersion}@${packageJson.version}`, '🟢', ''])
```
Comment (manudeli) [inline packages/react-query/src/scripts/utils/table.ts:38]:
```suggestion
  table.push(['export', `@suspensive/react-query-${currentVersion}@${targetPackageJson.version}`, '🟢', ''])
```

This should be not packageJson(@suspensive/react-query's package.json) targetPackageJson(@suspensive/react-query-4, @suspensive/react-query-5's package.json)

File: packages/react-query/src/scripts/utils/table.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  '<QueryErrorBoundary/>',
+]
+
+export function getStatusTable(currentVersion: string) {
+  const packageJson = getPackageJson()
+  const tanStackReactQueryPackageJson = getTanStackReactQueryPackageJson()
+  const tanStackReactQueryMajorVersion = tanStackReactQueryPackageJson.version.split('.')[0]
+
+  const table = new Table({
+    head: ['', 'result', 'status', 'advice'],
```
Comment (manudeli) [inline packages/react-query/src/scripts/utils/table.ts:31]:
```suggestion
    head: [packageJson.name, 'result', 'status', 'advice'],
```

File: packages/react-query/src/scripts/utils/table.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    style: { head: [] },
+  })
+
+  console.log(tanStackReactQueryMajorVersion)
+
+  table.push(['version', `@suspensive/react-query@${packageJson.version}`, '🟢', ''])
```
Comment (manudeli) [inline packages/react-query/src/scripts/utils/table.ts:37]:
```suggestion
  table.push(['version', packageJson.version, '🟢', ''])
```

**Review by manudeli — COMMENTED** [review]
I left some comment to suggest better console.log

**Comment by gwansikk** [comment]
@manudeli Additionally, how about providing a link to the official documentation in the "You can use" advice?

like this:

image

File: packages/react-query/src/scripts/utils/table.ts (hunk 1 above)
Comment (gwansikk) [reply packages/react-query/src/scripts/utils/table.ts]: I added `getSuspensiveReactQueryPackageJson` to accurately retrieve the targetVersion.

- [feat(react-query): add getSuspensiveReactQueryPackageJson util](https://github.com/toss/suspensive/pull/1100/commits/ce9d72a744bfec5dc0028b62399f7cbaf9299317)

**Comment by manudeli** [comment]
> @manudeli Additionally, how about providing a link to the official documentation in the "You can use" advice?
> 
> like this:
> 
> image

Cool! then we must maintain that link almost forever. Do we have to make redirector to make persistent link after merging this Pull Request?

**Comment by gwansikk** [comment]
> > @manudeli Additionally, how about providing a link to the official documentation in the "You can use" advice?
> > like this:
> > image
> 
> Cool! then we must maintain that link almost forever. Do we have to make redirector to make persistent link after merging this Pull Request?

Yes, if using that link. Alternatively, how about displaying it as a permanent link (e.g., GitHub or suspensive.org)? If you have any better ideas, please let me know.

**Comment by gwansikk** [comment]
@manudeli Ah, I understand! I've set it to "https://suspensive.org/links/react-query". Once the PR is merged, we need to quickly set up the permanent link.

image

File: packages/react-query/src/scripts/utils/table.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    currentTargetVersion === tanStackReactQueryMajorVersion ? '🟢' : '❌',
+    currentTargetVersion === tanStackReactQueryMajorVersion
+      ? 'The versions are compatible.'
+      : `Install @tanstack/react-query@${tanStackReactQueryMajorVersion} or\n execute suspensive-react-query switch ${tanStackReactQueryMajorVersion} to match\n @suspensive/react-query version with\n @tanstack/react-query`,
+  ])
+  table.push([
+    'You can use',
+    currentTargetVersion === '5' ? VERSION5_APIS.join('\n') : VERSION4_APIS.join('\n'),
+    '🟢',
+    'For more detailed information about the provided APIs,\nplease visit the official documentation:\nhttps://suspensive.org/links/react-query',
```
Comment (manudeli) [inline packages/react-query/src/scripts/utils/table.ts:55]:
```suggestion
    'For more detailed information about the provided APIs,\nplease visit the official documentation:\nhttps://suspensive.org/docs/react-query/motivation',
```

Could you do it next version? In my opinion, it's optional for this Pull Request.

**Review by manudeli — APPROVED** [review]

**Comment by manudeli** [comment]
Cool!

**Pull request merged**